sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -788,6 +790,24 @@ static Pair<DataSource, Filtration> getFiltration(
   {
     if (!canUseIntervalFiltering(dataSource)) {
       return Pair.of(dataSource, toFiltration(filter, virtualColumnRegistry.getFullRowSignature(), false));
+    } else if (dataSource instanceof UnnestDataSource) {
+      // UnnestDataSource can have another unnest data source
+      // join datasource, filtered data source, etc as base
+      Pair<DataSource, Filtration> pair = getFiltration(
+          ((UnnestDataSource) dataSource).getBase(),
+          filter,
+          virtualColumnRegistry,
+          joinableFactoryWrapper
+      );
+      return Pair.of(dataSource, pair.rhs);
+    } else if (dataSource instanceof FilteredDataSource) {
+      final FilteredDataSource filteredDataSource = (FilteredDataSource) dataSource;
+      final Filtration baseFiltration = Filtration.create(filteredDataSource.getFilter())
+                                                  .optimize(virtualColumnRegistry.getFullRowSignature());
```

Comment:
should we optimize optimizeFilterOnly if it is not using the intervals?

Reply:
This could be replaced by a call as follows 
```
final boolean status = canUseIntervalFiltering(filteredDataSource);
 final Filtration baseFiltration = toFiltration(filteredDataSource.getFilter(), virtualColumnRegistry.getFullRowSignature(), status);
      
```

Reply:
Updated this